c++/src/capnp/compat/http-over-capnp.c++

```diff
@@ -688,6 +831,88 @@ public:
     });
   }
 
+  kj::Promise<void> connect(ConnectContext context) override {
+    auto params = context.getParams();
+    auto host = params.getHost();
+    auto headers = factory.headersToKj(params.getHeaders());
+    auto upPipe = kj::newOneWayPipe();
+
+    PipelineBuilder<ConnectResults> pb;
+    pb.setUp(factory.streamFactory.kjToCapnp(kj::mv(upPipe.out)));
+
+    context.setPipeline(pb.build());
+
+    // This is similar to kj::HttpClientAdapter::connect.
+    auto pipe = kj::newTwoWayPipe();
+    auto paf = kj::newPromiseAndFulfiller<kj::HttpClient::ConnectRequest::Status>();
+    auto response = kj::refcounted<HttpOverCapnpConnectResponseImpl>(kj::mv(paf.fulfiller));
+
+    auto promise = inner->connect(host, headers, *pipe.ends[0], *response)
+        .eagerlyEvaluate([response=kj::mv(response),
+                          host=kj::mv(host),
+                          headers=kj::mv(headers),
+                          connectStream=kj::mv(pipe.ends[0])](kj::Exception&& ex) mutable {
+        response->handleException(kj::mv(ex), kj::mv(connectStream));
+      });
+
+    return paf.promise.then(
+        [this, upPipe = kj::mv(upPipe), down = params.getDown(), context = kj::mv(context),
+        io = kj::mv(pipe.ends[1])](auto status) mutable {
+      auto headers = status.headers->clone();
+
+      if (status.statusCode < 200 || status.statusCode > 299) {
+        auto req = context.getParams().getContext().startErrorRequest();
+        auto rpcResponse = req.initResponse();
+        rpcResponse.setStatusCode(status.statusCode);
+        rpcResponse.setStatusText(status.statusText);
+        rpcResponse.adoptHeaders(factory.headersToCapnp(
+            headers, Orphanage::getForMessageContaining(rpcResponse)));
+
+        // Set the body size if the error body exists.
+        auto bodySize = 0;
```

Comment:
This will declare `bodySize` to have type `int`, because that's the compiler's best guess at what type you intend `0` to have. Technically, though, HTTP response bodies can be multiple gigabytes, so this could overflow. You need to explicitly specify `uint64_t` here.

More abstractly speaking, initializing an `auto` from any literal value is weird. What is the type of, say, `123`? C language rules will tell you it is `int`, but it's kind of arbitrary. From a pure type theory perspective, the type of the literal `123` is "the set of values containing only a single member, which is the value 123". So from a theoretical perspective it looks like you're declaring a variable that cannot have any other value than the one you assigned...